test/TesterInternal/GeoClusterTests/MultiClusterRegistrationTests.cs

```diff
@@ -0,0 +1,346 @@
+            {
+                // logging  
+                foreach (var o in c.Overrides)
+                {
+                    o.Value.TraceLevelOverrides.Add(new Tuple<string, Severity>("Runtime.Catalog", Severity.Verbose));
+                    o.Value.TraceLevelOverrides.Add(new Tuple<string, Severity>("Orleans.GrainDirectory.LocalGrainDirectory", Severity.Verbose));
+                }
+            };
+
+            // Create 3 clusters, each with 2 silos. 
+            NewGeoCluster(globalserviceid, ClusterA, 2, addtracing);
+            NewGeoCluster(globalserviceid, ClusterB, 2, addtracing);
+            NewGeoCluster(globalserviceid, ClusterC, 2, addtracing);
+
+            WriteLog("Clusters are ready (elapsed = {0})", stopwatch.Elapsed);
+
+            WaitForLivenessToStabilizeAsync().WaitWithThrow(TimeSpan.FromMinutes(1));
+
+            // Create two clients per cluster.
+            ClientA0 = NewClient<ClientWrapper>(ClusterA, 0);
+            ClientA1 = NewClient<ClientWrapper>(ClusterA, 1);
+            ClientB0 = NewClient<ClientWrapper>(ClusterB, 0);
+            ClientB1 = NewClient<ClientWrapper>(ClusterB, 1);
+            ClientC0 = NewClient<ClientWrapper>(ClusterC, 0);
+            ClientC1 = NewClient<ClientWrapper>(ClusterC, 1);
+
+            ClientA0.InjectClusterConfiguration(ClusterA, ClusterB, ClusterC);
+            WaitForMultiClusterGossipToStabilizeAsync(false).WaitWithThrow(TimeSpan.FromMinutes(System.Diagnostics.Debugger.IsAttached ? 60 : 1));
+
+            Clients = new ClientWrapper[] { ClientA0, ClientA1, ClientB0, ClientB1, ClientC0, ClientC1 };
+            ClientClusters = new string[] { ClusterA, ClusterA, ClusterB, ClusterB, ClusterC, ClusterC };
+
+            stopwatch.Stop();
+            WriteLog("Clusters and clients are ready (elapsed = {0}).", stopwatch.Elapsed);
+        }
+
+        private ClientWrapper[] Clients;
+        private string[] ClientClusters;
+
+
+        Random random;
+        private int Next()
+        {
+            lock (random)
+                return random.Next();
+        }
+
+        // Kill all clients and silos.
+        public void Dispose()
+        {
+            WriteLog("Disposing...");
+            var stopwatch = new System.Diagnostics.Stopwatch();
+            stopwatch.Start();
+
+            try
+            {
+                var disposetask = Task.Run(() => StopAllClientsAndClusters());
+
+                disposetask.WaitWithThrow(TimeSpan.FromMinutes(System.Diagnostics.Debugger.IsAttached ? 60 : 2));
+            }
+            catch (Exception e)
+            {
+                WriteLog("Exception caught in test cleanup function: {0}", e);
+                throw e;
+            }
+
+            stopwatch.Stop();
+            WriteLog("Dispose completed (elapsed = {0}).", stopwatch.Elapsed);
+        }
+
+
+        [Fact, TestCategory("GeoCluster")]
+        public async Task All()
```

Comment:
I think the current version is sufficient for testing purposes  - there is no reason to run all tests if some tests fail, in fact I would prefer the test to finish as quickly as possible in that case. Also, all failing tests log their exceptions, so I can look at the log to see what was going on.